### PR TITLE
doc: Add DBT doc plugin

### PR DESCRIPTION
**Pull request by IIBenII**
## Hey, I just made a Pull Request!

Add DBT doc plugin in documentation.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: microsite/data/plugins/dbt.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: DBT Docs
+author: IIBenII
+authorUrl: https://github.com/IIBenII
+category: Discovery
+description: View DBT models and tests documentation.
+documentation: https://github.com/IIBenII/backstage-plugin-dbt
+iconUrl: https://avatars.githubusercontent.com/u/18339788?s=200&v=4
```
Comment (awanlin) [inline microsite/data/plugins/dbt.yaml:8]: Can you confirm that we can use this icon? It looks to be from dbt itself so we just need to be sure

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for the contribution @IIBenII, this looks pretty cool and well done. Just left a comment about the logo as we just need to be careful about those types of things :)

File: microsite/data/plugins/dbt.yaml (hunk 1 above)
Comment (IIBenII) [reply microsite/data/plugins/dbt.yaml]: Hi, I'm not sure about it, and documentation is confusing (https://www.getdbt.com/trademark-guidelines). By safety I removed the icon

**Review by awanlin — APPROVED** [review]

File: microsite/data/plugins/dbt.yaml (hunk 1 above)
Comment (awanlin) [reply microsite/data/plugins/dbt.yaml]: Thanks, sorry about that.

**Pull request closed without merging**